cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -159,12 +160,12 @@ static CompositeFuture run(Vertx vertx, KubernetesClient client, PlatformFeature
                     prom.handle(res);
                 });
         }
-        return CompositeFuture.join(futures);
+        return CompositeFuture.join(new ArrayList<>(futures));
```

Comment:
Why do we need to create a new list here?

Reply:
<img width="587" alt="截屏2021-04-05 02 47 29" src="https://user-images.githubusercontent.com/27606234/113518510-5572b000-95b9-11eb-83ff-6a51b51fe32d.png">

Because `CompositeFuture.join` required type `List<Future>`

Reply:
So why not keep it as `List<Future>` in the first place? It seems like a better solution than this.

Reply:
Generic types shouldn't be used raw (without type parameters) in variable declarations or return values. Doing so bypasses generic type checking, and defers the catch of unsafe code to runtime.

Reply:
The signature of `CompositeFuture.join()` using the raw type `Future` is the root cause of the problem, and there isn't much we can do about that, since it's not our code. It forces a raw type to be used _somewhere_. Using `new ArrayList<>(futures)` just lets the compiler infer the type argument to `ArrayList` to be the raw type `Future`, so there's still a raw type, it's just you didn't write it. It comes with the cost of allocating a pointless list, and it this case it doesn't actually make the code any more type safe. (I agree with the general point of avoiding raw types where possible, the problem is that, as here, it's not always possible.)

Perhaps the best thing to do would to be add the following method to `Util`:

```
    @SuppressWarnings({"unchecked", "rawtypes"})
    static <T> CompositeFuture join(List<Future<T>> futures) {
        return CompositeFuture.join((List) futures);
    }
```

and use it everywhere we currently call `CompositeFuture.join`. This isolates the use of raw types to a single method, so avoiding `@SuppressWarnings` at all the call sites.

Reply:
@tombentley Ok, i see

Reply:
> This isolates the use of raw types to a single method, so avoiding @SuppressWarnings at all the call sites.

There do not seem to be any `@SuppressWarnings` triggered by this. So would it be better to just try to address this at the source? (in Vert.x)

Reply:
@scholzj See eclipse-vertx/vert.x#3595 why Vert.x doesn't have this.